e2e/browser/testApp/components/accessGrants/index.tsx

```diff
@@ -0,0 +1,157 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import {
+  getDefaultSession
+} from "@inrupt/solid-client-authn-browser";
+import {
+  approveAccessRequest,
+  revokeAccessGrant,
+} from "@inrupt/solid-client-access-grants";
+import {
+  getPodUrlAll,
+  saveFileInContainer,
+  getSourceUrl,
+  deleteFile,
+} from "@inrupt/solid-client";
+import { useState } from "react";
+import React from "react";
+
+
+const session = getDefaultSession();
+const SHARED_FILE_CONTENT = "Some content.\n";
+
+export default function AccessGrant() {
+  const [accessGrant, setAccessGrant] = useState<string>();
+  const [sharedResourceIri, setSharedResourceIri] = useState<string>();
+
+  const handleCreate = (e) => {
+    // This prevents the default behaviour of the button, i.e. to resubmit, which reloads the page.
+    e.preventDefault();
+    if (typeof sharedResourceIri === "string") {
+      // If a resource already exist, do nothing
+      return;
+    }
+    (async () => {
+      if(typeof session.info.webId !== "string") {
+        console.error("You must be authenticated to create a resource.")
+        return;
+      }
+      // Create a file in the resource owner's Pod
+      const resourceOwnerPodAll = await getPodUrlAll(session.info.webId);
+      if (resourceOwnerPodAll.length === 0) {
+        throw new Error(
+          "The Resource Owner WebID Profile is missing a link to at least one Pod root."
+        );
```

Comment:
Ditto here, just throwing isn't particularly great, and we should report the status in the test UI ideally.

Reply:
94c0d8df7e83df384763ba2a11b96e955d0efbc6